File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {

    int ans = 0;

    public void dfs(TreeNode root, int maxSeen)
    {
        if(root == null)
            return;

        if(root.val >= maxSeen)
        {
            ans++;
        }

        maxSeen = Math.max(maxSeen, root.val);

        dfs(root.left, maxSeen);
        dfs(root.right, maxSeen);
    }

    public int goodNodes(TreeNode root)
    {
        dfs(root, root.val);
        return ans;
    }
}

File: Data Structures & Algorithms/count-number-of-islands/submission-0.py
class Solution:
    def numIslands(self, grid: List[List[str]]) -> int:
        ans = 0

        def dfs(i, j):
            if i < 0 or i >= len(grid) or j < 0 or j >= len(grid[0]):
                return

            if grid[i][j] == "0":
                return

            grid[i][j] = "0"

            dfs(i + 1, j)
            dfs(i, j + 1)
            dfs(i - 1, j)
            dfs(i, j - 1)

        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] == "1":
                    ans += 1
                    dfs(i, j)

        return ans